internal/segment-store/segment-store.go

```
@@ -203,3 +213,14 @@ func (segmentstore *SegmentStore) Delete(key []byte) (bool, error) {
 
 	return true, nil
 }
+
+func (segmentstore *SegmentStore) PrepareForCompaction() SegmentId {
+	segmentstore.mu.Lock()
+	defer segmentstore.mu.Unlock()
+
+	activeSegmentId := segmentstore.activeSegment.id
+
+	segmentstore.OpenNewSegmentFile()
```

golangci-lint (1.64.8)

Error return value of `segmentstore.OpenNewSegmentFile` is not checked

(errcheck)